### ISSUE TITLE
policy_gradient.py's problems

### ISSUE THREAD

**Issue by huangqingyi-code** [MarSaKi/nasnet#1]
hello.
    I have two questions in policy_gradient.py to ask you.
first:
def cal_loss(self, actions_p, actions_log_p, worker, baseline):
        reward = worker.acc - baseline
        policy_loss = -1 * torch.sum(actions_log_p * reward)
        entropy = -1 * torch.sum(actions_p * actions_log_p)
        entropy_bonus = -1 * entropy * self.entropy_weight
what does entory mean?
second:
            for episode in range(self.episodes):
                worker = results_queue.get()
                # worker.actions_p = torch.Tensor(worker.actions_p).to(self.device)
                worker.actions_index = torch.LongTensor(worker.actions_index).to(self.device)
                workers.append(worker)

worker.actions_index = torch.LongTensor(worker.actions_index).to(self.device) ,the worker.actions_index does't change?

**Comment by MarSaKi** [comment]
1. entropy measures the uncertainty of the decision, I suggest you refer to the original paper of "PPO".
2. Yeah, actions_index doesn't change. Each worker corresponds to an architecture.

Best Regards,

**Comment by huangqingyi-code** [comment]
Thank you for your reply,I have two more questions.
1.
for worker in workers:
     actions_p, actions_log_p = self.controller.get_p(worker.actions_index)
     loss += self.cal_loss(actions_p, actions_log_p, worker, self.baseline)
above have the: actions_p, actions_log_p, actions_index = self.controller.sample()
both of the actions_p and actions_log should be the same,because the actions_index(x) is same. why use the self.controller.get_p() again to get the actions_p
2.
controller.py's def sample():
input = torch.LongTensor([self.len_nodes + self.len_OPS]).to(self.device),
the input how to set?random?I think it is very important?
